samples/sensor/dht_polling/src/main.c

```diff
@@ -8,17 +8,35 @@
 #include <stdlib.h>
 
 #include <zephyr/device.h>
-#include <zephyr/drivers/sensor.h>
 #include <zephyr/sys/util_macro.h>
 #include <zephyr/kernel.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/sensor_data_types.h>
+#include <zephyr/rtio/rtio.h>
+#include <zephyr/dsp/print_format.h>
 
 #define DHT_ALIAS(i) DT_ALIAS(_CONCAT(dht, i))
-#define DHT_DEVICE(i, _)                                                                 \
+#define DHT_DEVICE(i, _)                                                      \
 	IF_ENABLED(DT_NODE_EXISTS(DHT_ALIAS(i)), (DEVICE_DT_GET(DHT_ALIAS(i)),))
 
 /* Support up to 10 temperature/humidity sensors */
 static const struct device *const sensors[] = {LISTIFY(10, DHT_DEVICE, ())};
 
+#define DHT_IODEV(i, _)                                                      \
+	IF_ENABLED(DT_NODE_EXISTS(DHT_ALIAS(i)),                                 \
+		(SENSOR_DT_READ_IODEV(_CONCAT(dht_iodev, i), DHT_ALIAS(i),           \
+		{SENSOR_CHAN_AMBIENT_TEMP, 0},                                       \
+		{SENSOR_CHAN_HUMIDITY, 0})))
+
+LISTIFY(10, DHT_IODEV, (;));
```

Comment:
I'm curious if there's a board/sample that has 10 DHT sensors

Reply:
I suppose a blame could tell us, I know one person had a ton of ADI 1wire to i2c adapters that subsequently ran to temperature sensors. Quite a lot of them in fact.

Reply:
Seems like #63016 :smile: